### PR TITLE
Workspace name sanitisation

### LINKED ISSUE #5661: Sanitising a workspace name while creating or editing
### Is your proposal related to a problem?

Currently, user can give a name to a workspace but the name has no defined properties which breaks the UI at some points. For example, the length of the name is defined as '25 characters' and because of this the UI breaks (check ss for reference)

### Describe the solution you'd like

Workspace admins should be able to create and edit workspace names. 

### Describe the Acceptance Criteria

In case of creating a new workspace and adding name to it : 
1) Name should be unique.
2) Name should be less than or equal to 40 alphanumeric characters with or without spaces.
3) Name should be mandatory. 
4) In case of same names or incorrect combinations, error handling should be added.

In case of editing workspace name : 
1) Name should be unique.
2) Name should be less than or equal to 40 alphanumeric characters with or without spaces.
3) Name should be mandatory. 
4) In case of same names or incorrect combinations, error handling should be added.
5) Name should get updated to the new one after changing it.

Screenshot 2023-03-01 at 7 28 30 PM

### PR DESCRIPTION
**Pull request by shah21**
Resolves #5661

### CODE REVIEW

File: server/migrations/1683136077244-AddUniqueConstraintToWorkspaceName.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      'organizations',
+      new TableUnique({
+        name: 'name_organizations_unique',
+        columnNames: ['name'],
+      })
+    );
+  }
+
+  public async migrateWorkspaceNames(entityManager: EntityManager) {
+    const workspaces = await entityManager.find(Organization);
```
Comment (gsmithun4) [inline server/migrations/1683136077244-AddUniqueConstraintToWorkspaceName.ts:18]: we don't want to get all workspaces and iterate 
```
select sub_query.name from (select count(*) as name_count, name from organizations group by name) sub_query where name_count > 1
```
use this query to get names which repeats and iterate the array

File: server/migrations/1683136077244-AddUniqueConstraintToWorkspaceName.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { Organization } from 'src/entities/organization.entity';
+import { EntityManager, MigrationInterface, QueryRunner, TableUnique } from 'typeorm';
+
+export class AddUniqueConstraintToWorkspaceName1683136077244 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const entityManager = queryRunner.manager;
+    await this.migrateWorkspaceNames(entityManager);
+    await queryRunner.createUniqueConstraint(
```
Comment (gsmithun4) [inline server/migrations/1683136077244-AddUniqueConstraintToWorkspaceName.ts:8]: can declare this in entity also

File: server/src/services/organizations.service.ts [hunk 3]
```diff
@@ -66,6 +66,12 @@ export class OrganizationsService {
   async create(name: string, user?: User, manager?: EntityManager): Promise<Organization> {
     let organization: Organization;
     await dbTransactionWrap(async (manager: EntityManager) => {
+      const ifWorkspaceNameExists = await manager.findOne(Organization, { name });
+
+      if (ifWorkspaceNameExists) {
+        throw new BadRequestException('Workspace name already exists.');
```
Comment (gsmithun4) [inline server/src/services/organizations.service.ts:72]: Should throw `ConflictException`

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: frontend/src/_helpers/handle-response.js [hunk 4]
```diff
@@ -28,3 +28,13 @@ export function handleResponseWithoutValidation(response) {
     return data;
   });
 }
+
+export const handleErrConnections = (error) => {
```
Comment (gsmithun4) [inline frontend/src/_helpers/handle-response.js:32]: shall we do this for all APIs using some global configs, we can do it in different PR

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: frontend/src/_helpers/handle-response.js (hunk 4 above)
Comment (shah21) [reply frontend/src/_helpers/handle-response.js:32]: Sure. I will remove this Part

**Review by gsmithun4 — APPROVED** [review]
